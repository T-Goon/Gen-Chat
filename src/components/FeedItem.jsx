import React from "react";
import { View, Text, StyleSheet } from "react-native";

const FeedItem = ({ username, message }) => {
    return (
        <View style={styles.container}>
            <Text style={styles.username}>{username+':'}</Text>
            <Text style={styles.message}>{message}</Text>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        backgroundColor: 'hsl(230, 100%, 80%)',
        marginVertical: 5,
        padding: 5,
        borderRadius: 10
    }, 
    username: {
        fontWeight: 'bold',
        fontSize: 20
    },
    message: {
        paddingLeft: 20
    }
});

export default FeedItem;