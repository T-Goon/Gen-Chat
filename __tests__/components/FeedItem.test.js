import React from 'react';
import { render } from '@testing-library/react-native';

import FeedItem from '../../src/components/FeedItem';

describe('<FeedItem/ >', () => {
    it('Num Children', () => {
        const feedItem = render(<FeedItem />);
        const json = feedItem.toJSON();

        expect(json.children.length).toBe(2);
    });

    it('Text Props Rendered', () => {
        const { getByText } = render(<FeedItem username='Username' message='Hellow!!' />);
        const username = getByText('Username:');
        const message = getByText('Hellow!!');

        expect(username).not.toBeNull();
        expect(message).not.toBeNull();
    });

    it('Snapshot', () => {
        const feedItemJson = render(<FeedItem username='Username' message='Hellow!!' />).toJSON();

        expect(feedItemJson).toMatchSnapshot();
    });
})