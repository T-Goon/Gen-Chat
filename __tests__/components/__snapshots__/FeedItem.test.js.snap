// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedItem/ > Snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "hsl(230, 100%, 80%)",
      "borderRadius": 10,
      "marginVertical": 5,
      "padding": 5,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Username:
  </Text>
  <Text
    style={
      Object {
        "paddingLeft": 20,
      }
    }
  >
    Hellow!!
  </Text>
</View>
`;
